### ISSUE TITLE
Standalone ContentRepository.Core !

### ISSUE THREAD

**Issue by bwaidelich** [neos/neos-development-collection#4227]
- [x] Get rid of `ObjectManagerInterface` in `NodeType` and `NodeTypeManager`
- [x] Replace `\Neos\Flow\Utility\Algorithms::generateUUID()` in ID-ValueObjects
- [x] Remove/rewrite Behat test traits!?
- [x] Make unit tests extend `\PHPUnit\Framework\TestCase` instead of `\Neos\Flow\Tests\UnitTestCase`
- [x] Get rid of Doctrine DBAL (v2) dependency (#4989)
- [ ] SQLite support
- [ ] Expose projection DBAL Schema (#4991)
- [ ] Remove `neos/eventstore-doctrineadapter` dependency (no hard requirement)

~- [ ] Improve `NodeTypeManager` API (#4228)~ Postponed to 9.1 or later – this can be done in a bridge layer

**Comment by bwaidelich** [comment]
`NodeType` and `NodeTypeManager` adjustments are related to #4228 but maybe we can already fix the Flow dependency so we are not blocked by the "rewrite"

**Comment by skurfuerst** [comment]
@nezaniel is working on this I guess :) or at least he has :)

**Comment by nezaniel** [comment]
#4455 should take care of the test suite dependencies

**Comment by mhsdesign** [comment]
Hi what is the status on this? I think we are mostly done with this right?

**Comment by mhsdesign** [comment]
I think it got an answer to it in https://github.com/skurfuerst/standalone-content-repository/tree/main ;)

We apparently still use Flows Algorithms https://github.com/skurfuerst/standalone-content-repository/blob/040d38f93950ca8fb028609c2ec03ca116ceb568/src/Workaround/FlowUtilityAlgorithms.php#L20 but that should be easy to adjust ^^


Also we must always declare `'Neos.ContentRepository:Root'` as nodetype manually, maybe thats the only nodetype the manager should always know about?

https://github.com/skurfuerst/standalone-content-repository/blob/040d38f93950ca8fb028609c2ec03ca116ceb568/NodeTypes.yaml#L3C1-L3C30

**Comment by skurfuerst** [comment]
@mhsdesign yeah I am thinking along the same lines :):)

**Comment by mhsdesign** [comment]
Okay it seems our logic in here is still valid:

https://github.com/neos/flow-development-collection/blob/e60dab42b10b09a0a23fbe463c40562b10f2baeb/Neos.Flow/Classes/Utility/Algorithms.php#L37-L44

`Ramsey\Uuid` will not use by its own `uuid_create`, only if the feature is enabled (`new FeatureSet(enablePecl: true)`):

The question now is where to place this logic and if the core should have an utility and a dependency on `ramsey/uuid`?

All leftover usages:

- https://github.com/neos/neos-development-collection/blob/3fbf9fd5fa126d5dce2f6246fb23ee84a6ff4272/Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Projection/NodeRelationAnchorPoint.php#L33 (and also in the postgres adapter)
- https://github.com/neos/neos-development-collection/blob/087a0a6864714640434c267b731bee690b4765bb/Neos.ContentGraph.DoctrineDbalAdapter/Tests/Behavior/Features/Bootstrap/ProjectionIntegrityViolationDetectionTrait.php#L239

**Comment by mhsdesign** [comment]
Update it seems we already have and utility which is already used for example for `NodeAggregateId::create`

See 

`Neos\ContentRepository\Core\SharedModel\Id\UuidFactory::create`

https://github.com/neos/neos-development-collection/blob/93aae710510b207d87f542d0d4bd8852ed3b6581/Neos.ContentRepository.Core/Classes/SharedModel/Id/UuidFactory.php#L26

That was introduced lately with https://github.com/neos/neos-development-collection/pull/4251 and we could use it again

**Comment by bwaidelich** [comment]
I had closed this, but there are still some open tasks in order to make this thing work nicely
